test/utils/fixture.ts

```diff
@@ -10,20 +21,73 @@ export function setupStageWithFixture(
   stageName: string,
   fixtureName: string
 ): void {
-  const stagePath = path.join(rootDir, stageName);
-  shell.mkdir(stagePath);
+  const stagePath = getStagePath(stageName);
+  shell.mkdir('-p', stagePath);
   shell.exec(
     `cp -a ${rootDir}/test/${testDir}/fixtures/${fixtureName}/. ${stagePath}/`
   );
-  shell.ln(
-    '-s',
-    path.join(rootDir, 'node_modules'),
-    path.join(stagePath, 'node_modules')
-  );
+  if (getPackageManager() === 'npm') {
+    shell.ln(
+      '-s',
+      path.join(rootDir, 'node_modules'),
+      path.join(stagePath, 'node_modules')
+    );
+  }
   shell.cd(stagePath);
+  if (getPackageManager() === 'yarn2') {
+    // Setup stage directory as a project root for yarn2
+    fs.writeFileSync(
+      '.yarnrc.yml',
+      `yarnPath: ${path.join(
+        rootDir,
+        'test/yarn2/.yarn/releases/yarn-sources.cjs'
+      )}\ncacheFolder: ${rootDir}/test/yarn2/.yarn/cache`
+    );
+    // Copy seed lockfile with most resolutions already performed.  Makes tests faster.
+    shell.exec(`cp ${rootDir}/test/yarn2/yarn.lock ${stagePath}/`);
+  }
+  packageManagerInstall();
```

Comment:
The tests are running `tsdx` within yarn pnp, which means each stage directory needs to have its dependencies resolved.  With running the stages in parallel, they each need to have their own PnP resolutions instead of sharing with the project root.  Another option could be creating all `stage-*` directories at once, declaring them as workspaces of the root, and running `yarn` once to resolve them all.

This approach, running `yarn` in each stage directory, is still pretty fast thanks to the seed `yarn.lock` and the pre-populated cache.  `yarn` essentially checks the `yarn.lock`, sees that it already knows the full dependency tree, sees that it already has all modules in cache, and can stop there.